osquery/tables/sleuthkit/sleuthkit.cpp

```diff
@@ -285,26 +285,22 @@ MultiHashes hashInode(TskFsFile* file) {
 
   // Allocate some heap memory and iterate over reading a chunk and updating.
   auto buffer_size = (size < 4096) ? size : 4096;
-  auto* buffer = (char*)malloc(buffer_size * sizeof(char));
-  if (buffer != nullptr) {
-    ssize_t chunk_size = 0;
-    for (ssize_t offset = 0; offset < size; offset += chunk_size) {
-      // Here max represents the local max requested bytes.
-      auto max = (size - offset < buffer_size) ? (size - offset) : buffer_size;
-      chunk_size =
-          file->read(offset, buffer, max, (TSK_FS_FILE_READ_FLAG_ENUM)0U);
-      if (chunk_size == -1 || chunk_size != max) {
-        // Huge problem, either a read failed or didn't read the max size.
-        free(buffer);
-        delete meta;
-        return MultiHashes();
-      }
-
-      md5.update(buffer, chunk_size);
-      sha1.update(buffer, chunk_size);
-      sha256.update(buffer, chunk_size);
+  char buffer[buffer_size];
```

Comment:
But it also increases the stack size. 
thread_local char *buffer = new char[buffer_size];
The above statement should address both the problems.

Reply:
But it will increase process memory footprint, will not it?

Reply:
It  does increase the size of process by  the multiple of number of thread executing this path.

This path is  executed by single thread at present. Using thread_local is for "just in case".

Big size stack variables are not recommended.  It can lead to unexpected crashes due to thread running out of stack. 

Your proposal to replacing the malloc with any other solution should be around  -
1. How often this code path is being executed.
2. how many cycles are you saving in a single allocation and free cycles. 

The suggestion which I gave is also useless, if this code path is not often executed then there is no point of keeping this amount of runtime memory engaged.